### PR TITLE
Allow multiple app instances and guard cross-process database schema migrations

### PR DESCRIPTION
**Pull request by jschick04**
## Summary

Fixes a regression where the app could no longer open more than one instance, and hardens the database schema-migration paths that concurrent instances can now reach.

### Part 1: Multi-instance launching
The handwritten WinUI entry point (`Program.Main`) was registering the process as single-instance via `AppInstance.FindOrRegisterForKey` and redirecting every subsequent launch into the first instance. Removing that restores per-launch windows:
- A separate right-click / double-click / "Open With" launches a **new** window.
- A multi-select right-click still opens the selection together in **one** new window (the shell extension already batches the selection into a single launch).
- Drag-and-drop remains the only path that adds logs to an existing window.

### Part 2: Cross-process schema-migration guards
With concurrent instances now possible, the SQLite schema paths needed cross-process coordination. Two primitives were added in `EventLogExpert.Logging.Concurrency`:
- `InterProcessMutex` - a named mutex for same-user (medium-integrity) coordination. `FileLogSink` is refactored onto it, and it guards the filter-library schema DDL.
- `InterProcessFileLock` - a sentinel-file lock that works across integrity levels (governed by file DACLs, not the mandatory integrity label), so a medium-IL app instance and the high-IL elevated helper coordinate on provider-database schema create/upgrade.

`ProviderDbContext` splits its upgrade probe into a lock-free core (used inside the held upgrade lock) and a public lock-taking probe (external readers), routes every schema entry point through the file lock, and refuses to recreate tables over a crashed mid-upgrade backup. A transient lock-timeout surfaces as `SchemaLockTimeoutException`, handled as a retryable/skip outcome by every caller rather than misclassifying the database.

### Testing
- 5 new unit tests for the concurrency primitives (log-mutex name byte-parity, cross-instance exclusion, abandoned-mutex recovery, timeout semantics, never-deleted sentinel).
- Full existing suite green: 2000+ tests across Logging, Eventing, Provider(.Database), Runtime, DatabaseTools, and the ElevationHelper integration tests (which exercise the file lock through a genuinely separate elevated process). MAUI head compiles.

### CODE REVIEW

File: src/EventLogExpert.Runtime/FilterLibrary/FilterLibrarySqliteStore.cs [hunk 1]
```diff
@@ -595,13 +650,7 @@ private async Task<SqliteConnection> OpenConnectionAsync(CancellationToken cance
...
             await connection.OpenAsync(cancellationToken).ConfigureAwait(false);
 
-            await using (var schemaCmd = connection.CreateCommand())
-            {
-                schemaCmd.CommandText = CreateTableSql;
-                await schemaCmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
-            }
-
-            await EnsureSchemaColumnsAsync(connection, cancellationToken).ConfigureAwait(false);
+            EnsureSchemaInitialized(connection);
```
[the remark this answers is not in the document]
Comment (jschick04) [reply src/EventLogExpert.Runtime/FilterLibrary/FilterLibrarySqliteStore.cs]: Good catch - the async->sync schema-init refactor dropped the token. Fixed in 603cfc62: `EnsureSchemaInitialized` now takes the `CancellationToken` and calls `ThrowIfCancellationRequested()` at entry and again right before each `InitializeSchema`, so a cancelled open no longer proceeds into the DDL work.

File: src/EventLogExpert.Logging/Concurrency/InterProcessMutex.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        {
+            throw new InvalidOperationException(
+                $"Interprocess mutex '{_name}' is unavailable; refusing to run guarded work unsynchronized.");
+        }
+
+        if (!TryRun(timeout, action))
+        {
+            throw new TimeoutException(
+                $"Timed out after {timeout.TotalSeconds:F0}s acquiring interprocess mutex '{_name}'.");
+        }
```
[the remark this answers is not in the document]
Comment (jschick04) [reply src/EventLogExpert.Logging/Concurrency/InterProcessMutex.cs:48]: Fixed in 603cfc62 - the timeout message now formats with `TotalSeconds:0.###`, so a sub-second timeout (e.g. 300ms) reads as `0.3s` instead of `0s`.

File: src/EventLogExpert.Logging/Concurrency/InterProcessFileLock.cs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    public void Run(TimeSpan timeout, Action action)
+    {
+        ArgumentNullException.ThrowIfNull(action);
+
+        if (!TryRun(timeout, action))
+        {
+            throw new TimeoutException(
+                $"Timed out after {timeout.TotalSeconds:F0}s acquiring interprocess file lock '{_lockPath}' (scope '{_scope}').");
+        }
```
[the remark this answers is not in the document]
Comment (jschick04) [reply src/EventLogExpert.Logging/Concurrency/InterProcessFileLock.cs:41]: Fixed in 603cfc62 - same `TotalSeconds:0.###` formatting change as the mutex, so sub-second lock-acquisition timeouts render meaningfully in diagnostics.

File: src/EventLogExpert.Logging/Concurrency/InterProcessFileLock.cs [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                $"Timed out after {timeout.TotalSeconds:0.###}s acquiring interprocess file lock '{_lockPath}' (scope '{_scope}').");
+        }
+    }
+
+    public bool TryRun(TimeSpan timeout, Action action)
+    {
+        ArgumentNullException.ThrowIfNull(action);
+
+        var deadline = Environment.TickCount64 + (long)timeout.TotalMilliseconds;
+        var backoff = s_minBackoff;
```
[the remark this answers is not in the document]
Comment (jschick04) [reply src/EventLogExpert.Logging/Concurrency/InterProcessFileLock.cs:57]: Fixed in f2b93846. TryRun now validates the timeout on the original TimeSpan: Timeout.InfiniteTimeSpan waits indefinitely (the retry loop skips the deadline check via an `infinite` flag), and any other negative value throws `ArgumentOutOfRangeException`. Validating on the raw `TimeSpan` (rather than the truncated `(long)TotalMilliseconds`) also rejects fractional negatives like `-1.5ms` and `-1 tick` that a millisecond cast would otherwise misclassify as infinite/try-once. Added a wait-then-acquire test for the infinite case plus a Theory covering those negative edge cases.

**Review by NikTilton — APPROVED** [review]
